services/frontend-v3/src/components/employeeSummary/officialLanguage/OfficialLanguage.jsx

```diff
@@ -22,43 +22,31 @@ const OfficialLanguage = ({ data }) => {
   };
 
   const getSecondLanguageGradeInfo = (dataSource) => {
-    const reading = dataSource.secondLangProfs.find(
-      (i) => i.proficiency === "READING"
-    );
+    const reading = dataSource.secondLangProfs
+      ? dataSource.secondLangProfs.find((i) => i.proficiency === "READING")
+      : null;
```

Comment:
To keep consistency, I would return `undefined` instead of `null`, since if the `.find()` function doesn't find something that satisfies the expression inside the function, it would return `undefined`. Same thing for the other paces below
```suggestion
      : undefined;
```